tests/Linter/CachingLinterTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Linter;
+
+use org\bovigo\vfs\vfsStream;
+use PhpCsFixer\Linter\CachingLinter;
+use PhpCsFixer\Tests\TestCase;
+
+/**
+ * @author ntzm
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Linter\CachingLinter
+ */
+final class CachingLinterTest extends TestCase
+{
+    /**
+     * @param bool $isAsync
+     *
+     * @dataProvider provideIsAsyncCases
+     */
+    public function testIsAsync($isAsync)
+    {
+        $sublinter = $this->prophesize('PhpCsFixer\Linter\LinterInterface');
+        $sublinter->isAsync()->willReturn($isAsync);
+
+        $linter = new CachingLinter($sublinter->reveal());
+
+        $this->assertSame($isAsync, $linter->isAsync());
+    }
+
+    public function provideIsAsyncCases()
+    {
+        return array(
+            array(true),
+            array(false),
+        );
+    }
+
+    public function testLintFileIsCalledOnceOnSameContent()
+    {
+        $fs = vfsStream::setup('root', null, array(
+            'foo.php' => '<?php echo "baz";',
+            'bar.php' => '<?php echo "baz";',
+            'baz.php' => '<?php echo "foobarbaz";',
+        ));
+
+        $result = $this->prophesize('PhpCsFixer\Linter\LintingResultInterface');
+
+        $sublinter = $this->prophesize('PhpCsFixer\Linter\LinterInterface');
+        $sublinter->lintFile($fs->url().'/foo.php')->shouldBeCalledTimes(1)->willReturn($result->reveal());
+        $sublinter->lintFile($fs->url().'/bar.php')->shouldNotBeCalled();
+        $sublinter->lintFile($fs->url().'/baz.php')->shouldBeCalled()->willReturn($result->reveal());
+
+        $linter = new CachingLinter($sublinter->reveal());
+
+        $linter->lintFile($fs->url().'/foo.php');
```

Comment:
Please add assertions on the result of `lintFile()` calls here and `lintSource()` in the following test.

Reply:
Will do, thanks 👍